qiskit_neko/tests/machine_learning/test_neural_network_classifier_primitives.py

```diff
@@ -0,0 +1,55 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for quantum neural networks classifier."""
+import numpy as np
+from ddt import ddt, data
+from qiskit.algorithms.optimizers import COBYLA
+from qiskit.primitives import Sampler as ReferenceSampler
+from qiskit.utils import algorithm_globals
+from qiskit_aer.primitives import Sampler as AerSampler
+from qiskit_machine_learning.algorithms.classifiers import VQC
+
+from qiskit_neko import decorators
+from qiskit_neko.tests import base
+
+
+@ddt
+class TestNeuralNetworkClassifierOnPrimitives(base.BaseTestCase):
+    """Test adapted from the qiskit_machine_learning tutorials."""
+
+    def setUp(self):
+        super().setUp()
+        if hasattr(self.backend.options, "seed_simulator"):
+            self.backend.set_options(seed_simulator=42)
+
+        self.samplers = dict(reference=ReferenceSampler(), aer=AerSampler())
+
+    @decorators.component_attr("terra", "backend", "machine_learning")
+    @data("reference", "aer")
+    def test_neural_network_classifier(self, implementation):
+        """Test the execution of quantum neural networks using OpflowQNN"""
+        rng = np.random.default_rng(seed=42)
+        algorithm_globals.random_seed = 42
```

Comment:
Is there any potential issue with global state here since we're changing a global seed?

Reply:
Do you mean
> algorithm_globals.random_seed = 42

If yes, then it is required by `VQC` as it samples initial point randomly and this line fixes reproducibility.

Reply:
I meant more are they any potential side effects because this is a global and potentially shared with other tests. But I guess this is just how seeds are set for algorithms stuff for better or worse and if there are any other algorithms tests that aren't setting a seed they should work with any seed.

Reply:
Yes, you are totally right.